WalletWasabi/Stores/BlockFilterSqliteStorage.cs

```diff
@@ -0,0 +1,243 @@
+using Microsoft.Data.Sqlite;
+using NBitcoin;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using WalletWasabi.Backend.Models;
+
+namespace WalletWasabi.Stores;
+
+/// <summary>
+/// Sqlite-based storage for block filters.
+/// </summary>
+/// <remarks>The implementation is not thread-safe because <see cref="SqliteConnection"/> is not thread-safe.</remarks>
+/// <seealso href="https://learn.microsoft.com/en-gb/dotnet/standard/data/sqlite/async">
+/// Async is not supported at the moment, so no async methods were used in this implementation.
+/// </seealso>
+public class BlockFilterSqliteStorage : IDisposable
+{
+	private bool _disposedValue;
+
+	private BlockFilterSqliteStorage(SqliteConnection connection)
+	{
+		Connection = connection;
+	}
+
+	/// <remarks>Connection cannot be accessed from multiple threads at the same time.</remarks>
+	private SqliteConnection Connection { get; }
+
+	/// <summary>
+	/// Opens a new sqlite connection to the given database file.
+	/// </summary>
+	/// <param name="path">Path to the sqlite database file.</param>
+	/// <param name="startingFilter">Starting filter to put into the filter table if the table needs to be created.</param>
+	/// <exception cref="InvalidOperationException">If there is an unrecoverable error.</exception>
+	/// <seealso href="https://dev.to/lefebvre/speed-up-sqlite-with-write-ahead-logging-wal-do">Write-ahead logging explained.</seealso>
+	public static BlockFilterSqliteStorage FromFile(string path, FilterModel? startingFilter = null)
+	{
+		SqliteConnection connection = new($"Data Source={path}");
```

Comment:
I wouldn't call it path because basically it is not a path but a connection string, that means that it can containg many things. For example, we could implement encryption simply by using:

```
Data Source=./mydb.db;Password={the-password};
```

Much better examples:
https://www.connectionstrings.com/sqlite/

Reply:
It's a path here. I would suggest to use `SqliteConnectionStringBuilder` for escaping.